### ISSUE TITLE
Unable to launch app in iOS simulator on M1 Mac (Apple Silicon Chip machines) Since Flutter 2.5.0

### ISSUE THREAD

**Issue by JoeyTeng** [AgoraIO/Agora-Flutter-SDK#424]
**Describe the bug**
Unable to launch app in iOS simulator on M1 Mac (Apple Silicon Chip machines) Since Flutter 2.5.0.

**To Reproduce**
1. Create an empty project, add `agora_rtc_engine: ^4.0.6` into `pubspec.yaml`
2. Run `flutter run -d <the-device-id-to-the-iOS-simulator> --debug`
3. See error

**Desktop (please complete the following information):**
- Device: Mac mini (M1, 2020)
 - OS: macOS Big Sur 11.4 (20F71)
 - Browser: Not Relavant
 - Version: Not sure what to put here.

**Smartphone (please complete the following information):**
 - Device: iPhone 12 Pro Max Simulator
 - OS: iOS 14.5 (18E182)
 - Browser: Not Relavant
 - Version: com.apple.CoreSimulator.SimRuntime.iOS-14-5 (simulator)

**Additional context**

Focus on these 2 lines please:

```log
ld: building for iOS Simulator, but linking in dylib built for iOS, file '/path/to/this/project/ios/Pods/AgoraRtcEngine_iOS/AgoraAIDenoiseExtension.framework/AgoraAIDenoiseExtension' for architecture arm64
clang: error: linker command failed with exit code 1 (use -v to see invocation)
```

<details>
  <summary>`flutter run` log</summary>

```shell
$ flutter run -d <iOS Simulator Device ID> --debug
```

```log
Launching lib/main.dart on iPhone 12 Pro Max 3 in debug mode...
Running pod install...                                              4.1s
Running Xcode build...                                                  
Xcode build done.                                           13.2s
Failed to build iOS app
Error output from Xcode build:
↳
    objc[69914]: Class AMSupportURLConnectionDelegate is implemented in both /usr/lib/libauthinstall.dylib (0x1f3cdec10) and /System/Library/PrivateFrameworks/MobileDevice.framework/Versions/A/MobileDevice (0x129a102b8). One
    of the two will be used. Which one is undefined.
    objc[69914]: Class AMSupportURLSession is implemented in both /usr/lib/libauthinstall.dylib (0x1f3cdec60) and /System/Library/PrivateFrameworks/MobileDevice.framework/Versions/A/MobileDevice (0x129a10308). One of the two
    will be used. Which one is undefined.
    ** BUILD FAILED **


Xcode's output:
↳
    /Users/<user_name>/.pub-cache/hosted/pub.dartlang.org/agora_rtc_engine-4.0.6/ios/Classes/Base/Extensions.swift:121:22: warning: 'delay' is deprecated
                "delay": delay,
                         ^
    /Users/<user_name>/.pub-cache/hosted/pub.dartlang.org/agora_rtc_engine-4.0.6/ios/Classes/Base/RtcEngine.swift:645:31: warning: 'startAudioMixing(_:loopback:replace:cycle:)' is deprecated: use startAudioMixing(.., startPos)
    instead
            callback.code(engine?.startAudioMixing(params["filePath"] as! String, loopback: params["loopback"] as! Bool, replace: params["replace"] as! Bool, cycle: (params["cycle"] as! NSNumber).intValue))
                                  ^
    /Users/<user_name>/.pub-cache/hosted/pub.dartlang.org/agora_rtc_engine-4.0.6/ios/Classes/Base/RtcEngine.swift:691:31: warning: 'getAudioMixingDuration()' is deprecated: use getAudioMixingDuration:(NSString* _Nullable)filePath
    instead
            callback.code(engine?.getAudioMixingDuration()) {
                                  ^
    /Users/<user_name>/.pub-cache/hosted/pub.dartlang.org/agora_rtc_engine-4.0.6/ios/Classes/Base/RtcEngine.swift:729:31: warning: 'playEffect(_:filePath:loopCount:pitch:pan:gain:publish:)' is deprecated: use
    playEffect:filePath:loopCount:pitch:pan:gain:publish:startPos instead.
            callback.code(engine?.playEffect((params["soundId"] as! NSNumber).int32Value, filePath: params["filePath"] as? String, loopCount: (params["loopCount"] as! NSNumber).int32Value, pitch: (params["pitch"] as!
            NSNumber).doubleValue, pan: (params["pan"] as! NSNumber).doubleValue, gain: (params["gain"] as! NSNumber).doubleValue, publish: params["publish"] as! Bool))
                                  ^
    /Users/<user_name>/.pub-cache/hosted/pub.dartlang.org/agora_rtc_engine-4.0.6/ios/Classes/Base/RtcEngine.swift:787:31: warning: 'setLocalVoiceChanger' is deprecated: use setAudioEffectPreset or setVoiceBeautifierPreset instead.
            callback.code(engine?.setLocalVoiceChanger(AgoraAudioVoiceChanger(rawValue: (params["voiceChanger"] as! NSNumber).intValue)!))
                                  ^
    /Users/<user_name>/.pub-cache/hosted/pub.dartlang.org/agora_rtc_engine-4.0.6/ios/Classes/Base/RtcEngine.swift:791:31: warning: 'setLocalVoiceReverbPreset' is deprecated: use setAudioEffectPreset or setVoiceBeautifierPreset
    instead.
            callback.code(engine?.setLocalVoiceReverbPreset(AgoraAudioReverbPreset(rawValue: (params["preset"] as! NSNumber).intValue)!))
                                  ^
    /Users/<user_name>/.pub-cache/hosted/pub.dartlang.org/agora_rtc_engine-4.0.6/ios/Classes/Base/RtcEngine.swift:992:31: warning: 'startAudioRecording(_:sampleRate:quality:)' is deprecated: use startAudioRecordingWithConfig:config
    instead.
            callback.code(engine?.startAudioRecording(params["filePath"] as! String, sampleRate: (params["sampleRate"] as! NSNumber).intValue, quality: AgoraAudioRecordingQuality(rawValue: (params["quality"] as!
            NSNumber).intValue)!))
                                  ^
    /Users/<user_name>/.pub-cache/hosted/pub.dartlang.org/agora_rtc_engine-4.0.6/ios/Classes/Base/Extensions.swift:121:22: warning: 'delay' is deprecated
                "delay": delay,
                         ^
    /Users/<user_name>/.pub-cache/hosted/pub.dartlang.org/agora_rtc_engine-4.0.6/ios/Classes/Base/RtcEngine.swift:645:31: warning: 'startAudioMixing(_:loopback:replace:cycle:)' is deprecated: use startAudioMixing(.., startPos)
    instead
            callback.code(engine?.startAudioMixing(params["filePath"] as! String, loopback: params["loopback"] as! Bool, replace: params["replace"] as! Bool, cycle: (params["cycle"] as! NSNumber).intValue))
                                  ^
    /Users/<user_name>/.pub-cache/hosted/pub.dartlang.org/agora_rtc_engine-4.0.6/ios/Classes/Base/RtcEngine.swift:691:31: warning: 'getAudioMixingDuration()' is deprecated: use getAudioMixingDuration:(NSString* _Nullable)filePath
    instead
            callback.code(engine?.getAudioMixingDuration()) {
                                  ^
    /Users/<user_name>/.pub-cache/hosted/pub.dartlang.org/agora_rtc_engine-4.0.6/ios/Classes/Base/RtcEngine.swift:729:31: warning: 'playEffect(_:filePath:loopCount:pitch:pan:gain:publish:)' is deprecated: use
    playEffect:filePath:loopCount:pitch:pan:gain:publish:startPos instead.
            callback.code(engine?.playEffect((params["soundId"] as! NSNumber).int32Value, filePath: params["filePath"] as? String, loopCount: (params["loopCount"] as! NSNumber).int32Value, pitch: (params["pitch"] as!
            NSNumber).doubleValue, pan: (params["pan"] as! NSNumber).doubleValue, gain: (params["gain"] as! NSNumber).doubleValue, publish: params["publish"] as! Bool))
                                  ^
    /Users/<user_name>/.pub-cache/hosted/pub.dartlang.org/agora_rtc_engine-4.0.6/ios/Classes/Base/RtcEngine.swift:787:31: warning: 'setLocalVoiceChanger' is deprecated: use setAudioEffectPreset or setVoiceBeautifierPreset instead.
            callback.code(engine?.setLocalVoiceChanger(AgoraAudioVoiceChanger(rawValue: (params["voiceChanger"] as! NSNumber).intValue)!))
                                  ^
    /Users/<user_name>/.pub-cache/hosted/pub.dartlang.org/agora_rtc_engine-4.0.6/ios/Classes/Base/RtcEngine.swift:791:31: warning: 'setLocalVoiceReverbPreset' is deprecated: use setAudioEffectPreset or setVoiceBeautifierPreset
    instead.
            callback.code(engine?.setLocalVoiceReverbPreset(AgoraAudioReverbPreset(rawValue: (params["preset"] as! NSNumber).intValue)!))
                                  ^
    /Users/<user_name>/.pub-cache/hosted/pub.dartlang.org/agora_rtc_engine-4.0.6/ios/Classes/Base/RtcEngine.swift:992:31: warning: 'startAudioRecording(_:sampleRate:quality:)' is deprecated: use startAudioRecordingWithConfig:config
    instead.
            callback.code(engine?.startAudioRecording(params["filePath"] as! String, sampleRate: (params["sampleRate"] as! NSNumber).intValue, quality: AgoraAudioRecordingQuality(rawValue: (params["quality"] as!
            NSNumber).intValue)!))
                                  ^
    ld: building for iOS Simulator, but linking in dylib built for iOS, file '/path/to/this/project/ios/Pods/AgoraRtcEngine_iOS/AgoraAIDenoiseExtension.framework/AgoraAIDenoiseExtension' for architecture arm64
    clang: error: linker command failed with exit code 1 (use -v to see invocation)
    note: Using new build system
    note: Building targets in parallel
    note: Planning build
    note: Analyzing workspace
    note: Constructing build description
    note: Build preparation complete
    warning: Stale file '/path/to/this/project/build/ios/Debug-iphonesimulator/Runner.app/Frameworks/Toast.framework' is located outside of the allowed root paths.

    warning: Stale file '/path/to/this/project/build/ios/Debug-iphonesimulator/Runner.app/Frameworks/fluttertoast.framework' is located outside of the allowed root paths.

    /path/to/this/project/ios/Pods/Pods.xcodeproj: warning: The iOS Simulator deployment target 'IPHONEOS_DEPLOYMENT_TARGET' is set to 8.0, but the range of supported deployment target
    versions is 9.0 to 14.5.99. (in target 'AgoraRtcEngine_iOS' from project 'Pods')

Could not build the application for the simulator.
Error launching application on iPhone 12 Pro Max 3.
```
</details>

**Comment by LichKing-2234** [comment]
duplicate of #425

**Comment by nilsreichardt** [comment]
@LichKing-2234 Is there no better fix than changing the Podfile as you mentioned in https://github.com/AgoraIO/Agora-Flutter-SDK/pull/425#issuecomment-920632271?
